### ISSUE TITLE
Bug for NullConditionalAssertion

### ISSUE THREAD

**Issue by iron9light** [fluentassertions/fluentassertions.analyzers#91]
```csharp
// cloudEvent.Time is Nullable<DateTime>
(cloudEvent.Time?.ToUniversalTime()).Should().Be(cloudEvent2.Time?.ToUniversalTime());
```
This code will get a NullConditionalAssertion warning.
But
```csharp
var x = cloudEvent.Time?.ToUniversalTime();
x.Should().Be(cloudEvent2.Time?.ToUniversalTime());
```
has no warning.

Is it a bug of the analyzer?
If not, how should I fix my code?

**Comment by thomaslevesque** [comment]
Just noticed the same thing. This is definitely a false positive.

**Comment by JeppeSN** [comment]
Please fix.

In an expression like:
```csharp
X?.Y.Should().Be(expected);
```
it is absolutely true that the entire "chain" `.Y.Should().Be(expected)` will be skipped if `X` is null. In particular, `Should` and `Be` are never called, and a warning is reasonable.

However, in the statement:
```csharp
(X?.Y).Should().Be(expected);
```
the chain stops at `)` and `Should` is always called. It is an extension method that can be called in a meaningful way on a null, even a reference-type null (which can occur if the compile-time type of `Y` is a reference type; in iron9light's example it was a value type which gives a `Nullable<>` which is also fine).

**Comment by Meir017** [comment]
@JeppeSN which analyzer is triggering the warning here?

**Comment by Meir017** [comment]
resolved by #170
